plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/util/LibraryElementHasher.java

```
@@ -0,0 +1,134 @@
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.emf.ecore.util.EcoreUtil.Copier;
+import org.eclipse.emf.ecore.xmi.XMLResource;
+import org.eclipse.emf.ecore.xmi.impl.XMIResourceFactoryImpl;
+import org.eclipse.fordiac.ide.model.emf.HashMetaData;
+import org.eclipse.fordiac.ide.model.libraryElement.INamedElement;
+import org.eclipse.fordiac.ide.ui.FordiacLogHelper;
+
+public final class LibraryElementHasher {
+	public static final String DEFAULT_HASH_ALG = "SHA3-512"; //$NON-NLS-1$
+	public static final String HASH_VERSION = "v1"; //$NON-NLS-1$ // increment this whenever anything is changed in the
+													// hashing algorithm
+	public static final String FULL_HASH_STRING = HASH_VERSION + ":" + DEFAULT_HASH_ALG; //$NON-NLS-1$
+
+	public static String getHashType() {
+		return FULL_HASH_STRING;
+	}
+
+	public static String getLibraryElementHash(final INamedElement libelem, final String version) {
+		final String[] versionParts = version.split(":"); //$NON-NLS-1$
+		if (versionParts.length != 2) {
+			return ""; // TODO raise exception maybe //$NON-NLS-1$
+		}
+		if (!versionParts[0].equals(HASH_VERSION)) {
+			return ""; // TODO raise exception maybe //$NON-NLS-1$
+		}
+		return hash(libelem, versionParts[1]);
+	}
+
+	private static String hash(final EObject eobj, final String version) {
+		MessageDigest digest = null;
+		try {
+			digest = MessageDigest.getInstance(version);
+		} catch (final NoSuchAlgorithmException e) {
+			FordiacLogHelper.logError("could not aquire hashing algorithm", e); //$NON-NLS-1$
+			return ""; //$NON-NLS-1$
```

I believe this should throw an exception instead of failing silently.

---

I was not sure. But I think you are right. Done